doc/overlays.xml

```diff
@@ -88,35 +88,37 @@ as <filename>overlays.nix</filename> and imported as the value of <literal>nixpk
 <title>Defining overlays</title>
 
 <para>Overlays are Nix functions which accept two arguments, 
-conventionally called <varname>self</varname> and <varname>super</varname>, 
+conventionally called <varname>final</varname> and <varname>previous</varname>,
 and return a set of packages. For example, the following is a valid overlay.</para>
 
 <programlisting>
-self: super:
+final: previous:
 
 {
-  boost = super.boost.override {
-    python = self.python3;
+  boost = previous.boost.override {
+    python = final.python3;
   };
-  rr = super.callPackage ./pkgs/rr {
-    stdenv = self.stdenv_32bit;
+  rr = previous.callPackage ./pkgs/rr {
+    stdenv = final.stdenv_32bit;
```

Comment:
it isn't really `final` though, is it? It can be overlayed on top of later.

Reply:
`final` is the composition of all overlays, including the future ones too.

However, the brevity and clarity of `final` in this `rr` example highlights that we should use e.g. `final.callPackage`, not `previous.callPackage` for functions, and use `previous` only when we explicitly want the _previous_ definition (because the current definition will shadow it in the final set).

Reply:
@grahamc `stdenv_32bit` is the final, as this is the result which went through the fix-point.
`rr` is not final yet, it might be overriden, and would be in the `previous` set of the next overlay.

Reply:
@orivej that makes sense to me. Updated PR to use `final.callPackage` unless @nbp disagrees and wants me to change it back.

Reply:
I disagree, `callPackage` is a function, and it should come from `previous`, not `final`.